warehouse/db.py

```diff
@@ -162,6 +162,12 @@ def cleanup(request):
         session.close()
         connection.close()
 
+    # Check if we're in read-only mode
+    from warehouse.admin.flags import AdminFlag
+    flag = session.query(AdminFlag).get('read-only')
+    if flag and flag.enabled:
```

Comment:
can we get an overall admin user short circuit here as well? that would alleviate the need to except for this in `edit_flag` and avoid having to except this for each and every operation an admin may need to perform during read-only situations.